src/equations/equations.jl

```diff
@@ -124,6 +125,47 @@ end
   return flux
 end
 
+# Dirichlet-type boundary condition for use with UnstructuredQuadMesh
+#  Note: For unstructured we lose the concept of a "direction"
+@inline function (boundary_condition::BoundaryConditionDirichlet)(u_inner, orientation, x, t,
+                                                                  surface_flux_function, equation)
+  # get the external value of the solution
+  u_boundary = boundary_condition.boundary_value_function(x, t, equation)
+
+  # Calculate boundary flux
+  flux = surface_flux_function(u_inner, u_boundary, orientation, equation)
+
+  return flux
+end
```

Comment:
```suggestion
@inline function (boundary_condition::BoundaryConditionDirichlet)(u_inner, normal::AbstractVector, x, t,
                                                                  surface_flux_function, equations)
  # get the external value of the solution
  u_boundary = boundary_condition.boundary_value_function(x, t, equations)

  # Calculate boundary flux
  flux = surface_flux_function(u_inner, u_boundary, normal, equation)

  return flux
end
```
We try to use `orientation::Integer` or `normal::AbstractVector` consistently, cf. https://trixi-framework.github.io/Trixi.jl/stable/meshes/tree_mesh/ and https://trixi-framework.github.io/Trixi.jl/stable/meshes/structured_mesh/